romancal/photom/photom.py

```diff
@@ -0,0 +1,169 @@
+
+    Raises
+    ------
+    Warning
+        When a field name is not in the table.
+    MatchFitsTableRowError
+        When more than one rows match.
+
+    Returns
+    -------
+    row : dict
+        Matched metadata keyword object
+    """
+
+    # Find photom entries that match the optical element in match_fields
+    results = [
+        item[1].upper() in dict(photom_table).keys() for item in
+        match_fields.items()
+    ]
+
+    # Trim to matched rows and test that we have exactly one match
+    photom_parameters_match = list(filter(bool, results))
+    if len(photom_parameters_match) > 1:
+        warnings.warn(f"Expected to find one matching row in table, found {len(photom_parameters_match)}.")
+    if len(photom_parameters_match) == 0:
+        warnings.warn("Expected to find one matching row in table, found 0.")
+        return None
+
+    # Return photom parameters for the selected optical element
+    return photom_table[match_fields['optical_element'].upper()]
+
+
+
+def save_area_info(input_model, photom_parameters):
+    """
+    Short Summary
+    -------------
+    Read the pixel area value in the photom parameters, then convert and
+    copy them to the metadata of the input datamodel.
+
+    Parameters
+    ----------
+    input_model : Roman level 2 image datamodel
+        input roman datamodel
+
+    photom_parameters : dict
+        Photom parameter object
+    """
+
+    # Load the average pixel area values from the photom reference file header
+    area_ster = photom_parameters["pixelareasr"]
+    area_a2 = photom_parameters["pixelareasr"] * ((u.arcsecond * u.arcsecond) / u.steradian) / A2_TO_SR
+
+    # Copy the pixel area values to the input model
+    log.debug('pixelarea_steradians = %s, pixelarea_arcsecsq = %s', str(area_ster), str(area_a2))
+    input_model.meta.photometry.pixelarea_arcsecsq = area_a2
+    input_model.meta.photometry.pixelarea_steradians = area_ster
+
+    # Return updated input model
+    return input_model
+
+
+def apply_photom(input_model, photom):
+    """
+    Short Summary
+    -------------
+    Open the reference file, retrieve the conversion factor from the reference
+    file that is appropriate to the instrument mode. The scalar factor is written
+    to the photmjsr and uncertainty keywords in the model.
+
+    For WFI, matching is based on optical_element.
+
+    Parameters
+    ----------
+    input_model : Roman level 2 image datamodel
+        input roman datamodel
+
+    photom : Roman Photom reference datamodel
+        photom roman datamodel
+
+    Returns
+    -------
+    output_model : Roman level 2 image datamodel
+        output roman datamodel with the flux calibration keywords updated
+    """
+
+    # Obtain photom parameters for the selected optical element
+    fields_to_match = {'optical_element': input_model.meta.instrument.optical_element}
+    photom_parameters = find_photom_parameters(photom.phot_table, fields_to_match)
```

Comment:
Does this work with the file format of `photom`?

Reply:
That function fully replaced with:
photom_parameters = photom.phot_table[input_model.meta.instrument.optical_element.upper()]

And the corresponding tests have been removed or repurposed.